### PR TITLE
Get gifsa articles

### PR DESCRIPTION
**Pull request by aiyus2020**

### CODE REVIEW

File: controllers/getArtGifController.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+const client = require("../models/db");
+const { getGif, getArts } = require("../queries/getArtGifQuery");
+class GetgifArticle {
+  async getGif(req, res) {
+    try {
+      const { gif_id } = req.params;
```
Comment (courage173) [inline controllers/getArtGifController.js:6]: this should be camel case

File: controllers/getArtGifController.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+      const result = await client.query(getGif, [gif_id]);
+
+      if (result.length === 0) {
+        // No GIF found with the specified ID
+        return res.status(404).json({ message: "GIF not found" });
+      }
+
+      res.json({
```
Comment (courage173) [inline controllers/getArtGifController.js:15]: this is wrong. you should write a query to fetch gifs and for each return 3 comments each.

File: queries/getArtGifQuery.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const client = require("../models/db");
+
+const getGif =
+  "SELECT g.gifs_id AS gif_id, c.gif_id AS comment_id, u.id, g.title, g.image_url,  c.comment AS gif_comment, g.created_at FROM upload g LEFT JOIN gif_comments c ON g.gifs_id = c.gif_id LEFT JOIN users u on g.gifs_id = u.id WHERE g.gifs_id = $1";
```
Comment (courage173) [inline queries/getArtGifQuery.js:4]: you have to use a subquery to select the comment so you can limit to 3 and also pass the limit as a parameter with a default of 3

**Review by courage173 — CHANGES_REQUESTED** [review]

File: controllers/getArtGifController.js (hunk 1 above)
Comment (aiyus2020) [reply controllers/getArtGifController.js:6]: done

File: controllers/getArtGifController.js (hunk 2 above)
Comment (aiyus2020) [reply controllers/getArtGifController.js:19]: done

File: queries/getArtGifQuery.js (hunk 3 above)
Comment (aiyus2020) [reply queries/getArtGifQuery.js]: done

**Review by courage173 — APPROVED** [review]

**Pull request merged**